syntax = "proto3";

package dev_observer.api.types.tokens;

import "google/protobuf/timestamp.proto";

enum AuthTokenProvider {
  UNKNOWN = 0;
  BIT_BUCKET = 1;
  JIRA = 2;
}

message AuthToken {
  string id = 1;
  string namespace = 2;
  AuthTokenProvider provider = 3;
  // Name of the workspace if this is a workspace token.
  optional string workspace = 4;
  // Full name of the repo if this is a repo token.
  optional string repo = 5;
  // True if this is a system auto-provisioned token (e.g. provided by Forge for BitBucket)
  bool system = 6;
  string token = 7;
  optional google.protobuf.Timestamp expires_at = 8;
  google.protobuf.Timestamp created_at = 9;
  google.protobuf.Timestamp updated_at = 10;
}

message TokensFilter {
  optional string namespace = 1;
  optional string workspace = 2;
}